src/routes/snippets.rs

```diff
@@ -21,6 +24,57 @@ fn create_snippet_impl(
     Ok(Created(location, Some(Output(new_snippet))))
 }
 
+fn create_link_header(
+    origin: &Origin,
+    next_marker: Option<String>,
+    prev_marker: Option<String>,
+    prev_needed: bool,
+) -> String {
+    let query_wo_marker = origin.query().map(|q| {
```

Comment:
Makes me wonder, should we support the case when `marker` is passed without actual value and hence sort-of invalid? E.g. `/v1/snippets?limit=1&marker`? Because checking `.starts_with("marker=")` implies there's always a value.

Reply:
Ah, good point! I guess, we could add a request guard for marker as well.